packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -99,22 +197,80 @@ class DomainColumn extends ColumnData<LinkData>
     VoidCallback? onPressed,
   }) {
     return _ErrorAwareText(
-      isError: dataObject.domainError,
+      isError: dataObject.domainErrors.isNotEmpty,
+      controller: controller,
       text: dataObject.domain,
+      link: dataObject,
     );
   }
+
+  // Default to show result with error first.
+  @override
+  int compare(LinkData a, LinkData b) {
+    final SortingOption? sortingOption =
+        controller.displayOptions.domainSortingOption;
+    if (sortingOption == null) return 0;
+
+    switch (sortingOption) {
+      case SortingOption.errorOnTop:
+        if (a.domainErrors.isNotEmpty) return 1;
+        if (b.domainErrors.isNotEmpty) return -1;
+        return 0;
+      case SortingOption.aToZ:
+        return a.domain.compareTo(b.domain);
+      case SortingOption.zToA:
+        return b.domain.compareTo(a.domain);
+    }
+  }
 }
 
 class PathColumn extends ColumnData<LinkData>
-    implements ColumnRenderer<LinkData> {
-  PathColumn()
+    implements ColumnRenderer<LinkData>, ColumnHeaderRenderer<LinkData> {
+  PathColumn(this.controller)
       : super(
           'Path',
           fixedWidthPx: scaleByFontFactor(kDeeplinkTableCellDefaultWidth),
         );
 
+  DeepLinksController controller;
+
   @override
-  bool get supportsSorting => true;
+  Widget? buildHeader(
+    BuildContext context,
+    Widget Function() defaultHeaderRenderer,
+  ) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        const Text('Path'),
+        PopupMenuButton<SortingOption>(
+          itemBuilder: (BuildContext context) {
+            return [
+              _buildPopupMenuSortingEntry(
+                controller,
+                SortingOption.errorOnTop,
+                isPath: true,
+              ),
+              _buildPopupMenuSortingEntry(
+                controller,
+                SortingOption.aToZ,
+                isPath: true,
+              ),
+              _buildPopupMenuSortingEntry(
+                controller,
+                SortingOption.zToA,
+                isPath: true,
+              ),
+            ];
+          },
+          child: Icon(
+            Icons.arrow_drop_down,
+            size: actionsIconSize,
+          ),
+        ),
+      ],
```

Comment:
this is duplicated above. Please extract to a helper widget that can be reused in both places